### PR TITLE
Add SEVIRI LST datareader

### PR DESCRIPTION
**Pull request by MWesselkamp**
## Description

This PR supersedes [#1803](https://github.com/ecmwf/WeatherGenerator/pull/1803) was based on `main`. This rebased version is now based on the latest `develop` as requested. Changes to 1803 include configuration files. 
Changes to 1803:
- updated configuration files

Note: The integration test `test_train_multi_stream` on the model directory currently fails.
The rest of the description follows 1803.

I implemented a custom datareader for the SEVIRI Level 2 land surface temperature product (LST). It works on on a gridded (lat lon) datastore of full Seviri images at hourly temporal resolution. The datastore is loaded with xarray during initialisation and lazily in the workers to avoid deadlocks caused by forked dataloaders.

The datareader extracts regions of interest from an image using indices that are stored in metadata files. It subsamples in time and space, e.g. here reduces hourly data to 6-hourly time steps. It loads normalisation statistics from external metadata files. It incorporates geoinfo (e.g. here DEM) from the datastore or returns empty geoinfo if not specified in the config.

This datareader is added to the MultiStreamSampler. A config used for finetuning for prognostic forecasting is added to configs/streams/.

I documented the datareader implementation with medium level of detail here: https://gitlab.jsc.fz-juelich.de/hedgedoc/s/LvbrmAWvf#

## Issue Number

This is not linked to an issue.

## Checklist before asking for review

-   [x] I have performed a self-review of my code
-   [x] My changes comply with basic sanity checks:
      - I have fixed formatting issues with `./scripts/actions.sh lint`
      - I have run unit tests with `./scripts/actions.sh unit-test`
      - I have documented my code and I have updated the docstrings.
      - I have added unit tests, if relevant
-   [x] I have tried my changes with data and code:
      - I have run the integration tests with `./scripts/actions.sh integration-test`
      - (bigger changes) I have run a full training and I have written in the comment the run_id(s): `launch-slurm.py --time 60`
      - (bigger changes and experiments) I have shared a hegdedoc in the github issue with all the configurations and runs for this experiments
-   [x] I have informed and aligned with people impacted by my change:
    - for config changes: the MatterMost channels and/or a design doc
    - for changes of dependencies: the MatterMost software development channel

### CODE REVIEW

File: packages/readers_extra/src/weathergen/readers_extra/data_reader_seviri.py [hunk 1]
```diff
@@ -0,0 +1,343 @@
+# (C) Copyright 2024 WeatherGenerator contributors.
```
Comment (clessig) [inline packages/readers_extra/src/weathergen/readers_extra/data_reader_seviri.py:1]: The data reader should go into the extra packages, i.e. `https://github.com/ecmwf/WeatherGenerator/tree/develop/packages/readers_extra`

File: src/weathergen/datasets/multi_stream_data_sampler.py [hunk 2]
```diff
@@ -155,6 +156,8 @@ def __init__(
                         dataset = DataReaderAnemoi
                     case "fesom":
                         dataset = DataReaderFesom
+                    case "msg_lst":
```
Comment (clessig) [inline src/weathergen/datasets/multi_stream_data_sampler.py]: Remove when the reader is in `https://github.com/ecmwf/WeatherGenerator/tree/develop/packages/readers_extra`

File: config/streams/seviri_lst/seviri_lst.yml [hunk 3]
```diff
@@ -0,0 +1,36 @@
+SEVIRI_LST :
```
Comment (clessig) [inline config/streams/seviri_lst/seviri_lst.yml]: Please remove from the PR. We need a separate repo for the configs and are in the process of creating it.

File: packages/readers_extra/src/weathergen/readers_extra/data_reader_seviri.py (hunk 1 above)
Comment (MWesselkamp) [reply packages/readers_extra/src/weathergen/readers_extra/data_reader_seviri.py:1]: done

File: src/weathergen/datasets/multi_stream_data_sampler.py (hunk 2 above)
Comment (MWesselkamp) [reply src/weathergen/datasets/multi_stream_data_sampler.py]: done

File: config/streams/seviri_lst/seviri_lst.yml (hunk 3 above)
Comment (MWesselkamp) [reply config/streams/seviri_lst/seviri_lst.yml]: done

**Comment by MWesselkamp** [comment]
Thanks, all comments were addressed. 
Note: The integration test that relies on the ..npp_radiation.zarr on leonardo failed.

File: packages/readers_extra/src/weathergen/readers_extra/registry.py [hunk 4]
```diff
@@ -24,5 +24,9 @@ def get_extra_reader(stream_type: str) -> object | None:
             from weathergen.readers_extra.data_reader_mesh import DataReaderMesh
 
             return DataReaderMesh
+        case "msg_lst":
```
Comment (pierluigicosi) [inline packages/readers_extra/src/weathergen/readers_extra/registry.py:27]: Do you need this name or can you use `case "seviri"` for consistency?

File: src/weathergen/datasets/multi_stream_data_sampler.py
Comment (pierluigicosi) [inline src/weathergen/datasets/multi_stream_data_sampler.py:1]: Seems there is some duplicate code here, unsure if we need it or it can be refactored.
